src/lazy/dsl.rs

```diff
@@ -831,6 +837,27 @@ impl JsExpr {
             .into()
     }
     #[napi(catch_unwind)]
+    pub fn str_json_extract(
+        &self,
+        dtype: Option<Wrap<DataType>>,
+        infer_schema_len: Option<i64>,
+    ) -> JsExpr {
+        let function = move |s: Series| {
+            let ca = s.utf8()?;
+            let dt = dtype.clone().map(|d| d.0 as DataType);
+            let infer_schema_len = infer_schema_len.map(|l| l as usize);
+            match ca.json_extract(dt, infer_schema_len) {
+                Ok(ca) => Ok(Some(ca.into_series())),
+                Err(e) => Err(PolarsError::ComputeError(format!("{:?}", e).into())),
+            }
+        };
+        self.clone()
+            .inner
+            .map(function, GetOutput::same_type())
+            .with_fmt("str.json_extract")
+            .into()
+    }
```

Comment:
polars `expr` has a json extract [function](https://github.com/pola-rs/polars/blob/9506d785ff221c30af836981b19285e7e25d182e/crates/polars-plan/src/dsl/string.rs#L492). We should be able to use that instead of using a closure.